src/compiler/tools/LanguageService.ts

```diff
@@ -230,6 +233,16 @@ export class LanguageService {
         return renameLocations.map(l => new RenameLocation(this.context, l));
     }
 
+    /**
+     * Gets the suggestion diagnostics.
+     * @param fileName - Optional source file to filter by.
+     */
+    getSuggestionDiagnostics(filePathOrSourceFile: SourceFile | string): DiagnosticWithLocation[] {
```

Comment:
Sorry, didn't notice this before. This should be in its own PR because it's a different feature. Doing so makes it work well with https://www.conventionalcommits.org (which is how I change the commit message when squashing the PR).

Thanks for adding this though!

Reply:
Actually, this is ok. I'll keep it in the PR because this is all language service changes.